### DISCUSSION TITLE
Zigbee plant/flower sensors for soil moisture + light + temperature + humidity product or kit?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#647] [General]
Does anyone other than my girlfriend believe that the world could really use a Zigbee based capacitive soil moisture + light (UV or lumen) + temperature and humidity sensor product designed for indoor houseplants home automation as well as look relatively good, is small and inexpensive?

Does anyone know of Zigbee device that could match all of most of those requirements?

Basically, we want is a Zigbee based alternative to the "**Xiaomi Mi Flora**" as MiFlora uses Bluetooth and has very poor signal reception.

https://www.home-assistant.io/integrations/miflora/

https://github.com/RodBr/miflora-card

https://www.domoticz.com/wiki/Mi_Flora_Bluetooth_LE

https://wiki.hackerspace.pl/projects:xiaomi-flora

**Xiaomi Mi Flora** smart plant sensor is sold rebranded under many other brands, inc. "Flower Care", "Grow Care Home" and "VegTrug".

Note! If buy it know _"There are “Chinese” and “International” versions available and there is a report that only the “International” works."_

https://www.xiaomiproducts.nl/en/xiaomi-mi-flower-care-plant-sensor.html

https://www.xiaomiproducts.nl/en/xiaomi-mi-flower-care-touch-sensor.html

![image](https://user-images.githubusercontent.com/6320001/105717435-6b605680-5f20-11eb-8a50-a32266c42876.png)

![image](https://user-images.githubusercontent.com/6320001/116196300-9cc10c00-a733-11eb-810f-169f37e2213c.png)

![image](https://user-images.githubusercontent.com/6320001/116198326-1d810780-a736-11eb-9962-4a6271072b0c.png)

[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/_jonF13th_4/0.jpg)](https://www.youtube.com/watch?v=_jonF13th_4)

https://www.youtube.com/watch?v=_jonF13th_4

**LilyGO TTGO HiGrow** is ESP32 WiFi-based but otherwise almost perfectly matches that description and is partially open source.
https://www.cnx-software.com/2020/11/10/ttgo-t-higrow-is-a-wifi-bluetooth-connected-soil-temperature-moisture-sensor/
http://www.lilygo.cn/prod_view.aspx?TypeId=50033&Id=1172&FId=t3:50033:3
https://www.aliexpress.com/item/32815782900.html
https://github.com/pesor/TTGO-T-HIGrow
https://github.com/Xinyuan-LilyGO/TTGO-HiGrow
https://github.com/LilyGO/higrowopen
https://github.com/LilyGO/HiGrow-Soil-testing

![image](https://user-images.githubusercontent.com/6320001/105716845-ae6dfa00-5f1f-11eb-9769-877cae218c21.png)

As far as I know, [Spruce Spoil Moisture sensor by Plaid Systems](https://spruceirrigation.com/sensors/) is the only Zigbee product that you can already buy which matches the description, but that is relatively large and expensive (plus it has an ugly colour and design). though its size and price are probably due to it being fully waterproof as design to outdoor use in your vegetable garden rather than potted plants indoors.

![image](https://user-images.githubusercontent.com/6320001/105716973-d65d5d80-5f1f-11eb-9443-8a2a08930e4b.png)

**DIYRuZ (modkam.ru) community** has the design for a Zigbee DIY version mentioned in https://github.com/zigpy/zigpy/discussions/646 but I for one would not mind seeing a version of that Zigbee flower sensor available as a complete kit with enclosure and/or an all finished product available for purchase.

DIYRuZ Soil Moisture Sensor Flower  = ZigbeeID: ["DIYRuZ_Flower"]
https://github.com/diyruz/flower
https://modkam.ru/?p=1700
https://zigbee.blakadder.com/DIYRuZ_Flower.html

![image](https://user-images.githubusercontent.com/6320001/105706626-0ef63a80-5f12-11eb-9adb-9f1b41ab5d22.png)

![image](https://user-images.githubusercontent.com/6320001/130772039-16481070-81c4-4d94-b7b9-e62e9a505120.png)

PS: I posted the same question in Home Assistant community forum but got no suggestions there as of yet https://community.home-assistant.io/t/zigbee-flower-sensors-for-soil-moisture-light-temperature-humidity/272803/

**Comment by jsiemek** [comment]
I use MiFlora sensors. Yes it's BLE but work OK even through walls. I use raspberry Pi Zero W as gateway. Not intuitive but works and pumps data via MQTT to HA. 

You can also simultaneously use phone app Flower Care.

Even if you needs more then one to cover whole area, Pi Zero W is cheap and I believe it will be simpler then building your own sensors.

**Reply by jsiemek** [reply]
Also watch this: https://youtu.be/udmJyncDvw0 before selecting soil sensor.

**Reply by Hedda** [reply]
> I use MiFlora sensors. Yes it's BLE but work OK even through walls. I use raspberry Pi Zero W as gateway. Not intuitive but works and pumps data via MQTT to HA. You can also simultaneously use phone app Flower Care. Even if you needs more then one to cover whole area, Pi Zero W is cheap and I believe it will be simpler then building your own sensors.

Note: Rant warning!

It is too bad that devices like MiFlora do not simply support standard Bluetooth Mesh Network based on the Bluetooth Low Energy specification, because think that would just be an excuse to buy more AC-powered devices that supports the same Bluetooth Mesh Network standard. Though I believe that there are not actually that many AC-powered devices that support standard Bluetooth Mesh Network? 

That is, I might be willing to jump on the bandwagon for Bluetooth Mesh Network if there were more AC-powered devices that supported it. Perhaps indoor BLE asset/people tracking could be an additional excuse for others but it is not something I have found interesting myself.

I prefer my home automation integrations to have as few controllers/gateways/bridges/hubs as possible and devices which use mesh networking standards like Zigbee and Z-Wave because in my experience that means less time on basic maintenance per device overall.

My scenario is based on that we currently live in a big house with four floors (if you count the basement and attic), with a huge chimney in the middle of the building which basically kills all radio signals, plus a large yard on top of that, so range and coverage is a challenge. Even if the signal passes through walls I know that would still need to install and maintain at least six such BLE gateways to achieve full coverage in our home today, and since all of those would probably use WiFi instead of wired ethernet they would, unfortunately, add to general signal interference on those frequencies. This is a hobby but we also have kids so as little time spent on maintenance after installation is a more important factor that the initial purchase cost of the actual devices, though I got to admit that I love a bargain, however my focus when buying devices is on more AC-powered products that help build out our Zigbee and Z-Wave Mesh Networks.

As it stands I today already have both a large Zigbee Mesh Network and a large Z-Wave Mesh Network, (as well as six WiFi Mesh Access-Points), and while we also have more home automation products that use other technologies I am a bit reluctant to jump on the bandwagon for a single purpose product that would require me to also maintain many BLE gateways as additional appliances as well.

**Reply by eth4n** [reply]
hi @jsiemek, what kind of software do you run on your raspberry to translate BLE to mqtt? plan to build up same setup (already got HA + mqtt running for zigbee2mqtt and a zero W is also somewhere around)!

**Reply by Hedda** [reply]
@eth4n please your question is off-topic for zigpy as BLE (Bluetooth) sensors like MiFlora has anything to do with Zigbee. Suggest that you post in the Home Assistant community forum instead (either start a new thread there or search if you can find an existing thread there that is on-topic but do not post about BLE sensors under their Zigbee section ) -> https://community.home-assistant.io

**Reply by eth4n** [reply]
@Hedda okay sorry for asking, just wanted to know the setup of @jsiemek who is using MiFlora with a PI to make data available. I just want to replicate this build as, as far as I can see, there is no good option for a zigbee based flower sensor...

**Comment by Hedda** [comment]
FYI, ITead has now released a wired external "Sonoff MS01 Smart Soil Moisture Sensor" addon that is meant to be paired with Sonoff TH10 or Sonoff TH16 (ESP8266 based devices) to monitor your plants soil moisture, but those make for very bulky and unpractical sensors for monitoring indoor potted plants like flowers, like how most people have in their windows. It also does not include other sensors like for; UV or lumen, temperature and humidity for the air.

https://itead.cc/product/sonoff-ms01-smart-soil-moisture-sensor/

https://www.youtube.com/watch?v=e5_bs0bvuag&ab_channel=ITEADStudio

![image](https://user-images.githubusercontent.com/6320001/130576546-ebf1720b-af61-4dd4-b119-e79bf0351457.png)

![image](https://user-images.githubusercontent.com/6320001/130576566-04c28cc6-a7dd-43be-8938-deb57b214840.png)

![image](https://user-images.githubusercontent.com/6320001/130576608-82af90f4-41fe-48f6-aa66-5c35d17c658f.png)

**Comment by Adminiuga** [comment]
> relatively good, is small and inexpensive?

You can only pick two out of those three requirements

**Reply by Hedda** [reply]
Then I would in this case pick **small** and **inexpensive** so that each device cost less than $20 since have many smaller pot plants.

So I guess same specification and quality as Xiaomi Mi Flora, but with a Zigbee MCU/radio instead of a Bluetooth MCU/radio.

The original Xiaomi Mi Flora is based on Dialog Semiconductor DA14580 SoC with BLE

Here is by the way a teardown, analysis and reverse engineering done on or for the Xiaomi Mi Flora plant monitor sensor:

https://wiki.hackerspace.pl/projects:xiaomi-flora

https://community.home-assistant.io/t/xiaomi-mi-plants-monitor-flower/3388

https://github.com/basnijholt/miflora

![image](https://user-images.githubusercontent.com/6320001/130759259-84015de5-7af9-4aa5-bd41-f56bd3fbffe1.png)

![image](https://user-images.githubusercontent.com/6320001/130759351-5a1e04cc-8984-4fae-b3a3-52d907499721.png)

**Comment by Hedda** [comment]
FYI, found this other "EFEKTA" DIY project based on nRF52840 which potentially could support Zigbee (via Nordic Semi nRF5 SDK)

https://www.openhardware.io/view/767/Wireless-Module-for-Capacitive-Soil-Moisture-Sensor-v20

https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE

https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE/tree/master/Zigbee

https://www.youtube.com/watch?v=LTzc0X-ESm4&feature=youtu.be&ab_channel=AndrewEfektaSB

![image](https://user-images.githubusercontent.com/6320001/130768467-930e2345-5f0d-4398-b571-31f9a4efd65e.png)

![image](https://user-images.githubusercontent.com/6320001/130768484-c6bbcd96-bdaf-4c2c-86c5-a0bd8e2c101a.png)

It uses this type of basic "capacitive soil moisture sensor" boards originally designed for Arduino:

https://www.aliexpress.com/item/32965744378.html

https://www.aliexpress.com/item/1005003107227825.html

https://www.aliexpress.com/item/4001131897353.html

https://www.aliexpress.com/item/32866171713.html

https://www.amazon.com/Capacitive-Moisture-Corrosion-Resistant-Detection/dp/B07SYBSHGX/

https://www.amazon.com/Treedix-Capacitive-Moisture-Detection-Watering/dp/B088FBF75L/

**Reply by Hedda** [reply]
> FYI, found this other "EFEKTA" DIY project based on nRF52840 which potentially could support Zigbee (via Nordic Semi nRF5 SDK)
> 
> https://www.openhardware.io/view/767/Wireless-Module-for-Capacitive-Soil-Moisture-Sensor-v20
> 
> https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE
> 
> https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE/tree/master/Zigbee
> 

Same dev also made a simpler Plant Watering Sensor Project for Zigbee but with CC2530 and based on the DIYRuZ Flower project

https://github.com/smartboxchannel/Plant-Watering-Sensor-Zigbee

https://www.tindie.com/products/diyberk/plant-watering-sensor-efekta-zigbee2mqtt/

https://www.youtube.com/watch?v=DzgeWvG4MEI&ab_channel=AndrewEfektaSB

![image](https://user-images.githubusercontent.com/6320001/130779313-e974bfd9-eb00-4e86-b23c-d0806a9ef4c3.png)

**Comment by Hedda** [comment]
FYI, stanvn has started development on a Zigbee plant watering sensor open source firmware project designed for b-parasite open source hardware based on the [Nordic Semi nRF5 / nRF52 SDK](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fstruct_sdk%2Fstruct%2Fsdk_nrf5_latest.html&cp=7_1) right now:

https://github.com/stanvn/zigbee-plant-sensor

In the Home Assistant community forum stanvn mention plans to create new modified hardware for it with more or other features:

https://community.home-assistant.io/t/zigbee-flower-and-plant-sensors-for-soil-moisture-light-temperature-humidity/272803/

The current b-parasite open source hardware is based on nRF52840 radio SoC and originally only has Bluetooth/BLE firmware, it currently features a resistive soil moisture sensor and other related plant monitor sensors such as lux and temperature/humidity meant for monitoring indoor flowers and potted plants need for water, sun light and heat.

https://github.com/rbaron/b-parasite

Other similar projects have previously all been based on TI’s CC2530:

https://github.com/smartboxchannel/Plant-Watering-Sensor-Zigbee

https://github.com/diyruz/flower

**Reply by Hedda** [reply]
> FYI, stanvn has started development on a Zigbee plant watering sensor open source firmware project designed for b-parasite open source hardware based on the [Nordic Semi nRF5 / nRF52 SDK](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fstruct_sdk%2Fstruct%2Fsdk_nrf5_latest.html&cp=7_1) right now:
> 
> https://github.com/stanvn/zigbee-plant-sensor

FYI, upstream b-parasite open-source soil moisture sensor hardware project will soon also get got Zigbee firmware support:

https://github.com/rbaron/b-parasite/pull/81

**Comment by krissrex** [comment]
I'm browsing around for this now. What is the current state of zigbee plant sensors in 2024?

**Reply by Hedda** [reply]
There is ongoing active discussion about both ready-made and DIY Zigbee-based products here:

* https://community.home-assistant.io/t/zigbee-flower-and-plant-sensors-for-soil-moisture-light-temperature-humidity/272803

Unfortunate there are still not any ”good” ready-made Zigbee products, but **b-parasite** with Zigbee firmware is a nice DIY project:

* https://github.com/rbaron/b-parasite
  * https://github.com/rbaron/b-parasite/tree/main/code/nrf-connect/samples/zigbee

PS: There also used to be a Russian making and selling a propitiatory and closed-source variant of the DIYRuZ Soil Moisture Sensor:

* https://github.com/diyruz/flower
  * https://www.zigbee2mqtt.io/devices/DIYRuZ_Flower.html
  * https://zigbee.blakadder.com/DIYRuZ_Flower.html
  * https://modkam.ru/?p=1700
